src/commands/Management/Helpers/guild-info.ts

```diff
@@ -0,0 +1,138 @@
+import { LanguageKeys } from '#lib/i18n/languageKeys';
+import { SkyraCommand, UserPaginatedMessage } from '#lib/structures';
+import type { GuildMessage } from '#lib/types';
+import { ZeroWidthSpace } from '#utils/constants';
+import { ApplyOptions } from '@sapphire/decorators';
+import { chunk } from '@sapphire/utilities';
+import { MessageEmbed, Permissions, Role } from 'discord.js';
+
+const SORT = (x: Role, y: Role) => Number(y.position > x.position) || Number(x.position === y.position) - 1;
+const roleMention = (role: Role) => role.toString();
+const roleLimit = 15;
+
+const paginatedMessagePermissions = new Permissions([Permissions.FLAGS.ADD_REACTIONS, Permissions.FLAGS.MANAGE_MESSAGES]);
+
+@ApplyOptions<SkyraCommand.Options>({
+	aliases: ['server-info'],
+	cooldown: 15,
+	description: LanguageKeys.Commands.Management.GuildInfoDescription,
+	extendedHelp: LanguageKeys.Commands.Management.GuildInfoExtended,
+	permissions: ['EMBED_LINKS'],
+	runIn: ['text']
+})
+export class UserCommand extends SkyraCommand {
+	public async run(message: GuildMessage, args: SkyraCommand.Args) {
+		const color = await this.context.db.fetchColor(message);
+		const roles = this.getRoles(args);
+
+		if (message.channel.permissionsFor(message.guild.me!)!.has(paginatedMessagePermissions)) {
+			const display = await this.buildDisplay(args, roles, color);
+			return display.start(message);
+		}
+
+		return message.send(await this.getSummary(args, roles, color));
+	}
+
+	private async buildDisplay(args: SkyraCommand.Args, roles: Role[], color: number): Promise<UserPaginatedMessage> {
+		const guild = args.message.guild!;
+		const display = new UserPaginatedMessage({
+			template: new MessageEmbed() //
+				.setColor(color)
+				.setThumbnail(guild.iconURL()!)
+				.setTitle(`${guild.name} [${guild.id}]`)
+		});
+
+		display.addPageEmbed(await this.getSummary(args, roles, color));
+
+		if (roles.length <= roleLimit) return display;
```

Comment:
So wait.. if there's less than 15 roles, you enumerate them, else you... enumerate AND list fully?

Reply:
Enumerate fully... in other pages. It's just useless to generate role pages if they're all in the summary.